error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -615,6 +615,58 @@ Flux<S> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Mono#flatMap(Function)} over more contrived alternatives. */
+  static final class MonoFlatMapIdentity<S, T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<S> mono, Function<S, Mono<T>> function) {
+      return mono.map(function).flatMap(identity());
+    }
+
+    @AfterTemplate
+    Mono<T> after(Mono<S> mono, Function<S, Mono<T>> function) {
+      return mono.flatMap(function);
+    }
+  }
+
+  /** Prefer {@link Mono#flatMapMany(Function)} over more contrived alternatives. */
+  static final class FlatMapManyIdentity<S, T> {
+    @BeforeTemplate
+    Flux<T> before(Mono<S> mono, Function<S, ? extends Publisher<T>> function) {
```

Comment:
Admittedly, I couldn't figure out why
```suggestion
    Flux<T> before(Mono<S> mono, Function<S, Publisher<T>> function) {
```
alone wouldn't do it. Anyone an idea what I'm missing here?

Reply:
In the latter case only function that return precisely a `Publisher` (so not a subtype) are matched.

Reply:
:bulb: Thanks a lot - clear.